script-test/playbackstrategies/modifiers/html5tests.js

```diff
@@ -0,0 +1,3260 @@
+              expect(player.getSource()).toBe(undefined);
+            });
+
+            it('Get Mime Type Returns Undefined In Empty State', function () {
+              expect(player.getMimeType()).toBe(undefined);
+            });
+
+            it('Get Current Time Returns Undefined In Empty State', function () {
+              expect(player.getCurrentTime()).toBe(undefined);
+            });
+
+            it('Get Seekable Range Returns Undefined In Empty State', function () {
+              expect(player.getSeekableRange()).toBe(undefined);
+            });
+
+            it('Get Duration Returns Undefined In Empty State', function () {
+              expect(player.getDuration()).toBe(undefined);
+            });
+
+            it('Get Source Returns Undefined In Empty State After Reset', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'testUrl', 'testMimeType', sourceContainer, {});
+              metaDataCallback();
+              finishedBufferingCallback();
+
+              player.reset();
+
+              expect(player.getSource()).toBe(undefined);
+            });
+
+            it('Get Mime Type Returns Undefined In Empty State After Reset', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'testUrl', 'testMimeType', sourceContainer, {});
+              metaDataCallback();
+              finishedBufferingCallback();
+
+              player.reset();
+
+              expect(player.getMimeType()).toBe(undefined);
+            });
+
+            it('Get Current Time Returns Undefined In Empty State After Reset', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'testUrl', 'testMimeType', sourceContainer, {});
+              metaDataCallback();
+              finishedBufferingCallback();
+
+              player.reset();
+
+              expect(player.getCurrentTime()).toBe(undefined);
+            });
+
+            it('Get Seekable Range Returns Undefined In Empty State After Reset', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'testUrl', 'testMimeType', sourceContainer, {});
+              metaDataCallback();
+              finishedBufferingCallback();
+
+              player.reset();
+
+              expect(player.getSeekableRange()).toBe(undefined);
+            });
+
+            it('Get Duration Returns Undefined In Empty State After Reset', function () {
+              player.initialiseMedia(MediaPlayerBase.TYPE.VIDEO, 'testUrl', 'testMimeType', sourceContainer, {});
+              metaDataCallback();
+              finishedBufferingCallback();
+
+              player.reset();
+
+              expect(player.getDuration()).toBe(undefined);
+            });
+
+            it('Calling Begin Playback In Empty State Is An Error', function () {
+              player.beginPlayback();
+
+              expect(logger.error).toHaveBeenCalledWith('Cannot beginPlayback while in the \'EMPTY\' state');
```

Comment:
You might want to assert on more than just the logger - when these types of error happen (bad API calls), we also change state to error, fire an event, and essentially reset the media player.

Same goes for all tests like this.